### DISCUSSION TITLE
How to categorise non-return control flow keywords?

### DISCUSSION THREAD

**Discussion by ghost** [nvim-treesitter/nvim-treesitter#4276] [Q&A] [answered]
I have an [open PR] (https://github.com/maxxnino/tree-sitter-zig/pull/22) to the Zig grammar that includes and categorises all Zig keywords in highlight groups. In writing it I had some difficulty categorising control flow keywords: do `break` and `continue` (standard meanings within loops) count as `@keyword.return`? I just barely came down on "yes" because they marked an exit from the current block.

Zig also has coroutines, controlled by the keywords `async` (initiate coroutine), `await` (suspend until coroutine completes and return result), `suspend` (save coroutine state then run attached block), `resume` (resume coroutine from last suspend point, return at next suspend point). There's also `nosuspend` which asserts that a given suspend point does not suspend (and hence does not make the containing function a coroutine). These are currently in `@keyword` but I really feel like they should be in a subgroup; which one would I put them in?

There are also `defer` and `errdefer`, which execute a statement on exit from a block (not a function), either unconditionally or on error. These also feel vaguely control-flow-y so maaaybe they should be in a subgroup, but I'm less uncomfortable with them being in `@keyword`.

What does everyone think?

**Comment by amaanq** [comment, marked as answer]
Solved by #4382
